packages/ui/src/DataTable/components/Actions/index.tsx

```diff
@@ -61,10 +67,30 @@ export const ActionsMenu = ({
         label: deleteLabel || "Delete",
         icon: deleteIcon || "pi pi-trash",
         className: "danger",
-        command: onDelete,
+        command: () => {
+          if (requireConfirmationOnDelete) {
+            setShowDeleteConfirmation(true);
+          } else {
+            console.log(onDelete);
+          }
+        },
       });
     }
   }
 
-  return <Menu model={items} buttonOptions={buttonOptions} />;
+  return (
+    <>
+      <Menu model={items} buttonOptions={buttonOptions} />
+      <ConfirmationModal
+        visible={showDeleteConfirmation}
+        message="Are you sure you want to delete?"
+        header="Confirm delete"
+        onHide={() => setShowDeleteConfirmation(false)}
+        accept={() => {
+          console.log(onDelete);
+          setShowDeleteConfirmation(false);
+        }}
```

Comment:
We cannot do console.log here. We should actually call onDelete with the row data where we clicked the delete action.